packages/components/src/DataTable/Column/DataTableCell.tsx

```diff
@@ -40,46 +41,64 @@ export interface DataTableCellProps
   size?: DataTableColumnSize
 }
 
-const DataTableCellLayout: FC<DataTableCellProps> = ({
-  children,
-  description,
-  indicator,
-  size,
-  ...props
-}) => {
-  let content =
-    size && size !== 'nowrap' ? <Truncate>{children}</Truncate> : children
-
-  if (description) {
-    content = (
-      <SpaceVertical gap="xxxsmall">
-        <span>{content}</span>
-        {description && (
-          <Paragraph fontSize="xsmall" color="subdued" truncate>
-            <Truncate>{description}</Truncate>
-          </Paragraph>
-        )}
-      </SpaceVertical>
-    )
+const DataTableCellLayout = forwardRef(
+  (props: DataTableCellProps, forwardedRef: Ref<HTMLElement>) => {
+    const { children, description, indicator, size } = props
+
+    let content =
+      size && size !== 'nowrap' ? <Truncate>{children}</Truncate> : children
+
+    const ref = useRef<HTMLTableDataCellElement>(null)
+    const forkedRef = useForkedRef(ref, forwardedRef)
 
-    if (indicator) {
+    useEffect(() => {
+      const element = ref?.current
+      const button = element?.querySelector('button')
+      const link = element?.querySelector('a')
+      const input = element?.querySelector('input')
+
+      button?.setAttribute('tabIndex', '-1')
+      link?.setAttribute('tabIndex', '-1')
+      input?.setAttribute('tabIndex', '-1')
```

Comment:
I agree with @lukelooker 's suggestion. This should be:
```suggestion
      if (element) {
        const focusableElements = Array.from(element.querySelectorAll('button,a,input'))
        focusableElements.forEach((el) => el.setAttribute('tabindex', '-1'))
      }
```